### PR TITLE
feat(orc8r): Add verbosity control to orc8r for lte/{network_id}/subscribers (fixes #3421)

### LINKED ISSUE #3421: Add verbosity option when listing a network's subscribers
Current  `/lte/{network_id}/subscribers `returns the list of subscribers with very verbose state information per subscriber. Add support for a non-verbose option that returns just the list of subscribers in the network, e.g., `/lte/{network_id}/subscribers?verbose=false`.

### PR DESCRIPTION
**Pull request by WugPro**
## Summary

fixes #3421

tl;dr: add a verbose query parameter to /lte/{network_id}/subscribers which defaults to true for existing behavior and when sets to false, returns only subscriber MSISDNs instead of full details.

## Test Plan

I have tested the modified endpoint from the swagger API page and using curl from the command line to verify that the endpoint handles the new parameter correctly and that it shows up in the documentation as expected.

Example output:

```
wug@wug-mbp test_certs % curl -X 'GET' \           
  'https://localhost:9443/magma/v1/lte/test/subscribers?verbose=true' \
  -H 'accept: application/json' --cert admin_operator.pem --key admin_operator.key.pem --cacert rootCA.pem -k
{"next_page_token":"","subscribers":{"IMSI208950000000010":{"config":{"lte":{"auth_algo":"MILENAGE","auth_key":"AAAAAAAAAAAAAAAAAAAAAA==","auth_opc":"AAECAwQFBgcICQoLDA0ODw==","state":"INACTIVE","sub_profile":"default"}},"id":"IMSI208950000000010","lte":{"auth_algo":"MILENAGE","auth_key":"AAAAAAAAAAAAAAAAAAAAAA==","auth_opc":"AAECAwQFBgcICQoLDA0ODw==","state":"INACTIVE","sub_profile":"default"},"name":"Jane Doe"}},"total_count":1}
wug@wug-mbp test_certs % curl -X 'GET' \
  'https://localhost:9443/magma/v1/lte/test/subscribers?verbose=false' \
  -H 'accept: application/json' --cert admin_operator.pem --key admin_operator.key.pem --cacert rootCA.pem -k
{"next_page_token":"","subscribers":["IMSI208950000000010"],"total_count":1}
```

Also verified that absence of verbose, and garbage verbose values, both work as expected (both should use verbose mode)

I have also added a test for this functionality to the existing tests for subscriber listing.

- [ ] This change is backwards-breaking

### CODE REVIEW

File: lte/cloud/go/services/subscriberdb/obsidian/handlers/handlers.go [hunk 1]
```diff
@@ -127,6 +128,23 @@ type subscriberFilter func(sub *subscribermodels.Subscriber) bool
 
 func acceptAll(*subscribermodels.Subscriber) bool { return true }
 
+// subscribersVerbosity filters a subscribers list for the specified verbosity.
+// If verbose mode is used, the list is returned as-is (a map of MSISDN strings
+// to Subscribers).
+// If non-verbose mode is used, an array of MSISDN strings is returned instead.
+func subscribersVerbosity(subs map[string]*subscribermodels.Subscriber, verbose bool) interface{} {
```
Comment (hcgatewood) [inline lte/cloud/go/services/subscriberdb/obsidian/handlers/handlers.go]: Nit: verbify func name: "makeVerbose" or "makeSubscribersVerbose"

File: lte/cloud/go/services/subscriberdb/obsidian/handlers/handlers_test.go [hunk 2]
```diff
@@ -434,6 +434,32 @@ func TestListSubscribers(t *testing.T) {
...
+		NextPageToken: expectedResult.NextPageToken,
+	}
+
+	for k := range expectedResult.Subscribers {
+		expectedResultAbbreviated.Subscribers = append(expectedResultAbbreviated.Subscribers, k)
+	}
+
+	if len(expectedResult.Subscribers) == 0 || expectedResult.TotalCount == 0 {
+		t.Errorf("Checking output schema of empty subscriber list? That's a problem!" +
+			"Make sure there's at least one subscriber.")
```
Comment (hcgatewood) [inline lte/cloud/go/services/subscriberdb/obsidian/handlers/handlers_test.go]: Can't tell how long this line is, but generally don't break unless it's 120 chars or even a bit more

File: lte/cloud/go/services/subscriberdb/obsidian/handlers/handlers_test.go [hunk 3]
```diff
@@ -434,6 +434,32 @@ func TestListSubscribers(t *testing.T) {
...
+	expectedResultAbbreviated := subscriberModels.PaginatedSubscribersAbbreviated{
+		TotalCount:    expectedResult.TotalCount,
+		NextPageToken: expectedResult.NextPageToken,
+	}
+
+	for k := range expectedResult.Subscribers {
+		expectedResultAbbreviated.Subscribers = append(expectedResultAbbreviated.Subscribers, k)
+	}
+
+	if len(expectedResult.Subscribers) == 0 || expectedResult.TotalCount == 0 {
```
Comment (hcgatewood) [inline lte/cloud/go/services/subscriberdb/obsidian/handlers/handlers_test.go]: Use `assert` pattern rather than `t.Errorf`

File: lte/cloud/go/services/subscriberdb/obsidian/models/swagger.v1.yml [hunk 4]
```diff
@@ -59,6 +61,11 @@ paths:
           type: string
           description: Filter to subscribers assigned the passed IP address
           required: false
+        - in: query
+          name: verbose
+          type: boolean
+          description: Show full subscribers if true (default) or just show IDs
+          required: false
```
Comment (hcgatewood) [inline lte/cloud/go/services/subscriberdb/obsidian/models/swagger.v1.yml]: (Should be able to) nix this section, since we handle it in the below "endpoint"

File: lte/cloud/go/services/subscriberdb/obsidian/models/swagger.v1.yml [hunk 5]
```diff
@@ -386,6 +425,27 @@ definitions:
           x-nullable: true
           $ref: '#/definitions/subscriber'
 
+  paginated_subscribers_abbreviated:
```
Comment (hcgatewood) [inline lte/cloud/go/services/subscriberdb/obsidian/models/swagger.v1.yml]: paginated_subscriber_ids

File: lte/cloud/go/services/subscriberdb/obsidian/models/swagger.v1.yml [hunk 6]
```diff
@@ -386,6 +425,27 @@ definitions:
...
+    description: Abbreviated page of subscribers
```
Comment (hcgatewood) [inline lte/cloud/go/services/subscriberdb/obsidian/models/swagger.v1.yml]: Page of subscriber IDs

File: lte/cloud/go/services/subscriberdb/obsidian/models/swagger.v1.yml [hunk 7]
```diff
@@ -386,6 +425,27 @@ definitions:
...
+    type: object
+    required:
+      - next_page_token
+      - subscribers
+      - total_count
+    properties:
+      total_count:
+        type: integer
+        format: int64
+        description: estimated total number of subscriber entries
```
Comment (hcgatewood) [inline lte/cloud/go/services/subscriberdb/obsidian/models/swagger.v1.yml]:
```suggestion
        description: Estimated total number of subscriber entries
```

**Review by hcgatewood — APPROVED** [review]
Looks great! Couple small requests, message me when you're ready and I'll merge

**Comment by WugPro** [comment]
Hang on a second.

File: lte/cloud/go/services/subscriberdb/obsidian/handlers/handlers.go [hunk 8]
```diff
@@ -127,6 +128,23 @@ type subscriberFilter func(sub *subscribermodels.Subscriber) bool
 
 func acceptAll(*subscribermodels.Subscriber) bool { return true }
 
+// filterSubscribersIfNonverbose filters a subscribers list for the specified verbosity.
+// If verbose mode is used, the list is returned as-is (a map of MSISDN strings
+// to Subscribers).
+// If non-verbose mode is used, an array of MSISDN strings is returned instead.
+func filterSubscribersIfNonverbose(subs map[string]*subscribermodels.Subscriber, verbose bool) interface{} {
```
Comment (hcgatewood) [inline lte/cloud/go/services/subscriberdb/obsidian/handlers/handlers.go]: Small request: I think "filter" is a misnomer here, because we're actually doing a conditional *map*, rather than *filter*. I'd go with `mapVerbose` or `mapSubscribersForVerbosity` or something like that

**Review by hcgatewood — COMMENTED** [review]
Not sure what the issue with the codegen is... maybe give it a day then rebase onto master. Msg after handling the 1 req then I'll merge for you! 🙂
